libs/category/src/resolvers/category-page-server-side/category-page-server-side.resolver.ts

```diff
@@ -0,0 +1,43 @@
+import { isPlatformBrowser } from '@angular/common'
+import { Inject, Injectable, PLATFORM_ID } from '@angular/core'
+import { ActivatedRouteSnapshot, Resolve } from '@angular/router'
+import { ActionsSubject, Store } from '@ngrx/store';
+import { Observable, of } from 'rxjs'
+import { filter, map, take } from 'rxjs/operators';
+
+import { DaffCategoryActionTypes, DaffCategoryLoad, DaffCategoryLoadFailure, DaffCategoryLoadSuccess } from '../../actions/category.actions';
+import { DaffCategoryReducersState } from '../../reducers/category-reducers.interface';
+import { DaffDefaultCategoryPageSize } from './default-category-page-size.token';
+
+/**
+ * Resolves the category page immediately when in a browser, so client-side loading states can show immediately. 
+ * When server-side, this resolver will wait until the category finishes loading to resolve the url mainly for seo considerations.
+ */
+@Injectable({
+	providedIn: 'root'
+})
+export class DaffCategoryPageServerSideResolver implements Resolve<Observable<boolean>> {
+  constructor(
+		@Inject(PLATFORM_ID) private platformId: string,
+		@Inject(DaffDefaultCategoryPageSize) private defaultCategoryPageSize: number,
+    private store: Store<DaffCategoryReducersState>,
+    private dispatcher: ActionsSubject,
+	) {}
+	
+	resolve(route: ActivatedRouteSnapshot): Observable<boolean> {
+		this.store.dispatch(new DaffCategoryLoad({
+      id: route.paramMap.get('id'), page_size: this.defaultCategoryPageSize
+		}));
+
+		return isPlatformBrowser(this.platformId) ? 
```

Comment:
Should the resolver care about whether its on the server or not? Should it be up to the app dev to selectively invoke it on the server vs the client? I feel like a resolver should be environment-agnostic.

Reply:
Or, at the very least, there should be an environment-agnostic resolver that this server-specific resolver can wrap and invoke.

Reply:
changed